### ISSUE TITLE
Document how to use a custom truststore

### ISSUE THREAD

**Issue by marschall** [spring-projects/spring-ldap#545]
If you want to use a custom truststore, eg. with just the root CA certificate of the server, you have to do: 

* implement a custom `SSLSocketFactory`
* implement a custom `SimpleDirContextAuthenticationStrategy` that sets the `java.naming.ldap.factory.socket` property in the `#setupEnvironment(Hashtable, String, String)` method.
* implement a custom `DefaultSpringSecurityContextSource` that in sets the `java.naming.ldap.factory.socket` property in the `#getAuthenticatedEnv(String, String)` method

It would be good if this was documented somewhere.

This is a follow up to #494

**Comment by ChristopherSchultz** [comment]
Yes, please! Difficult TLS configuration is something that is holding the whole industry back from becoming more secure. People resort to crazy things like modifying the JVM's `cacerts` file or setting JVM-wide trust stores instead of configuring each connection with separate trust. It's even worse when trying to provide client TLS certificates.

**Comment by rwinch** [comment]
Any interest in providing a PR?

**Comment by ChristopherSchultz** [comment]
I'm not an actual direct user of Spring, only a user of a product which uses it. I'm afraid I'm not steeped enough in Spring-isms to do a very good job with this.

The custom `SSLSocketFactory` is [the easy part](https://github.com/ChristopherSchultz/ssltest/blob/bb8bb0ae5b59c4d4eda9f37a712a42fc894daff8/src/java/net/christopherschultz/ssltest/SSLUtils.java#L158) (for me). The wiring into Spring is not.

**Comment by marschall** [comment]
> Any interest in providing a PR?

I can give it a try. Where would that go? `docs/asciidoc/index.adoc`? If so which chapter?

**Comment by rwinch** [comment]
Thanks for volunteering @marschall! A section in https://github.com/spring-projects/spring-ldap/blob/master/src/docs/asciidoc/index.adoc#configuration would probably be where I'd put it.

**Comment by JanHron** [comment]
@marschall What you've done is awesome and quite involved. Combined with the `CompositeX509TrustManager` available elsewhere, this gives us a very powerful toolkit for managing trust per individual connection.

**Comment by Hakky54** [comment]
I wanted to share here that it is easier to configure the ssl of spring boot as shown here: https://github.com/spring-projects/spring-ldap/issues/547#issuecomment-2269935433
